server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -69,26 +70,26 @@ public void setDomainList(DomainList domainList) {
         this.domainList = domainList;
     }
 
-    protected void isValidUsername(String username) throws UsersRepositoryException {
+    protected void validateUsername(String username) throws UsersRepositoryException {
         Username user = Username.of(username);
         if (supportVirtualHosting()) {
             // need a @ in the username
             if (!user.hasDomainPart()) {
-                throw new UsersRepositoryException("Given Username needs to contain a @domainpart");
+                throw new InvalidUsernameException("Given Username needs to contain a @domainpart");
             } else {
                 Domain domain = user.getDomainPart().get();
                 try {
                     if (!domainList.containsDomain(domain)) {
-                        throw new UsersRepositoryException("Domain does not exist in DomainList");
+                        throw new InvalidUsernameException("Domain does not exist in DomainList");
                     }
                 } catch (DomainListException e) {
-                    throw new UsersRepositoryException("Unable to query DomainList", e);
+                    throw new InvalidUsernameException("Unable to query DomainList", e);
```

Comment:
Please have a distinct exection for this one: it SHOULD the a `500`, we don't know if the name is valid or not!